src/test/scala/mesosphere/marathon/api/v2/PodsResourceTest.scala

```diff
@@ -943,6 +943,422 @@ class PodsResourceTest extends AkkaUnitTest with Mockito with JerseyTest {
+            |    "containers": [{
+            |        "name": "container0",
+            |        "resources": {
+            |            "cpus": 0.1,
+            |            "mem": 32
+            |        },
+            |        "image": {
+            |            "kind": "DOCKER",
+            |            "id": "private/image"
+            |        },
+            |        "linuxInfo": {
+            |          "ipcInfo": {
+            |              "mode": "SHARE_PARENT"
+            |          }
+            |        },
+            |        "exec": {
+            |            "command": {
+            |                "shell": "sleep 1"
+            |            }
+            |        }
+            |    }],
+            |    "linuxInfo": {
+            |       "ipcInfo": {
+            |           "mode": "PRIVATE"
+            |       }
+            |    }
+            |}
+          """.stripMargin
+
+        val response = asyncRequest { r =>
+          f.podsResource.create(podJson.getBytes(), force = false, f.auth.request, r)
+        }
+
+        withClue(s"response body: ${response.getEntity}") {
+          response.getStatus should be(201)
+        }
+      }
+
+      "Decline a pod definition with shared parent IPC and shm size defined" in {
+        implicit val podSystem = mock[PodManager]
+        val f = Fixture()
+
+        podSystem.create(any, eq(false)).returns(Future.successful(DeploymentPlan.empty))
+
+        val podJson =
+          """
+            |{
+            |    "id": "/pod",
+            |    "containers": [{
+            |        "name": "container0",
+            |        "resources": {
+            |            "cpus": 0.1,
+            |            "mem": 32
+            |        },
+            |        "image": {
+            |            "kind": "DOCKER",
+            |            "id": "private/image"
+            |        },
+            |        "linuxInfo": {
+            |          "ipcInfo": {
+            |              "mode": "SHARE_PARENT"
+            |          }
+            |        },
+            |        "exec": {
+            |            "command": {
+            |                "shell": "sleep 1"
+            |            }
+            |        }
+            |    }],
+            |    "linuxInfo": {
+            |        "ipcInfo": {
+            |            "mode": "SHARE_PARENT",
+            |            "shmSize": 16
+            |        }
+            |    }
+            |}
+          """.stripMargin
+
+        val response = asyncRequest { r =>
+          f.podsResource.create(podJson.getBytes(), force = false, f.auth.request, r)
+        }
+
+        withClue(s"response body: ${response.getEntity}") {
+          response.getStatus should be(422)
+          response.getEntity.toString should include("ipcInfo shmShize can NOT be set when mode is SHARE_PARENT")
+        }
+      }
+
+      "Accept a pod definition with shared Parent IPC and shm size defined" in {
```

Comment:
The purpose or impl of this test seem borked. There's a test
> Decline a pod definition with shared parent IPC and shm size defined
and this one,
> Accept a pod definition with shared Parent IPC and shm size defined

Both cannot be true I guess … and this one **does not** set shmSize, so maybe it should be
> Accept a pod definition with shared Parent IPC without shm size defined

?

Reply:
Yes, forgot to add the NOT in the test name, good catch.